### ISSUE TITLE
`chat_id` need to accept type int and string

### ISSUE THREAD

**Issue by DinoLeung** [DinoLeung/TeleDart#53]
## Describe the bug
Currently it only accept id(int), however according to the official docs, it needs to accept username(string) also

```
Unique identifier for the target chat or username of the target channel (in the format `@channelusername`)
```

**Comment by castaway** [comment]
Any likelyhood of applying the patch?

**Comment by DinoLeung** [comment]
@castaway I'm opposed to use `dynamic` in the library, as it beat the purpose of having it statically typed. I have looking into the [sealed_unions]( https://pub.dev/packages/sealed_unions) which does union type, still not decided to use it or not.

**Comment by castaway** [comment]
Fair enough, could always make a sendChatMessage, sendUserMessage ? Or always take a string, and convert to int when sending if it parses as one?

**Comment by DinoLeung** [comment]
@castaway changing function parameters type would be considered a breaking change, which means it would break all existing bots depends on this library.

 Although adding seperated implementation for each send message function would fix the issue we face, it also add unnecessary complexity to the project.

It is indeed a tough decision to make.

**Comment by DinoLeung** [comment]
#99 has a better idea
